src/v/cluster/persisted_stm.h

```diff
@@ -94,6 +94,7 @@ class persisted_stm
     virtual ss::future<> remove_persistent_state();
 
     ss::future<> make_snapshot();
+    uint64_t get_snapshot_size() const;
```

Comment:
Should this function be virtual as well? We "override" it in `rm_stm`, but not quite, so it can be confusing, which function is called (the caller can expect to call the function of the derived class, while in reality, the base class function will be called).